### ISSUE TITLE
Request: "Install only for me" option during installation process

### ISSUE THREAD

**Issue by bkidwell** [Open-Shell/Open-Shell-Menu#138]
See #136 . "Install only for me" or "Autostart at login only for me" would be very useful in multi-user environments where you don't want to surprised all the other users on the system.

**Comment by bkidwell** [comment]
Workaround: Extract the contents of the installer and do a copy-paste install to your preferred folder. Run "StartMenu.exe", edit and save settings. By default it will now automatically start for only the user that executed it.

**Comment by XenHat** [comment]
That should be relatively easy to implement. I don't really have time at the moment but we accept pull requests

**Comment by Moopere** [comment]
There are settings for this inside Classic Shell already.  I think its called "Auto start for this user" or somesuch.

When you install CS or OS, just ensure that this setting is turned off.  

I usually just capture  the whole registry key(s) pertaining to CS and roll that out with preferred settings for all users.  Those that want CS can individually turn it on themselves.

**Comment by bkidwell** [comment]
"Install only for me" is a common option for desktop customization software like this. I want to be sure when I install it on a SHARED server that I affect no one else but me -- no writing to Program Files and no writing to the global Start Menu.

**Comment by ibuprophen1** [comment]
This feature request is definitely a pending item as I would like to also have this option implemented. 

Please be patient regarding this feature. It's one of the top items on the "to-do" list (at the moment). 

As @XenHat had previously stated... 

For ANYONE who (of course, versed with Pull Requests on Github) has any ideas, suggestions, etc... 

We are more than happy to provide ANY Pull Requests regarding fixes, enhancements, features, etc... and, of course, if it works, it will be merged. 

Thank you very much for your time, patience, understanding and especially your support for this software! 👌 


~Ibuprophen

**Comment by bkidwell** [comment]
Yes @ibuprophen1 , I agree that it is important and low priority -- especially since my stated workaround is good. I didn't mean to imply any demand when I was replying to @Moopere .

**Comment by ibuprophen1** [comment]
No Worries! 🎉 

I do encourage the occasional feedback to all issues reported here. 

When i see issues that have no member feedback, contributions, status updates, etc... for a long time (approx 60+ days), it makes me think that the original OP member who reported the issue has either resolved it (in one fashion or another) OR may have no interest at all in the issue OR something else. 

I'm doing what I can to help the developmental team out with scrubbing through all these open issues (primarily the older ones) and apply the appropriate action as necessary (Closing, Tagging, etc..). 

I had planned to create a Pinned List of the various "to-do's" here so those of you can see what's done and not done but, more on this will be seen as soon as I can get a handle on the reported issues as explained above. 

WE REALLY appreciate everyone's patience and THANK YOU!!!! 🎂 🙃  


~Ibuprophen

**Comment by ge0rdi** [comment]
There are two official ways that allow disabling `Open-Shell` autostart by default, while retaining the possibility for particular users to enable it.

Please, check this comment:
https://github.com/Open-Shell/Open-Shell-Menu/issues/161#issuecomment-500226960

That's said, I think there is no need to do any changes in installer.

**Comment by danielmreck** [comment]
@ge0rdi, in addition to @bkidwell's original purpose to not "want to surprise all the other users on the system," there is a better reason to improve the installer to allow a user to install only for themselves.

Many of us are working on corporate machines where "Install for All User" privileges are disabled.  At present, these users cannot install Open Shell for themselves at all.  The suggestions above to disable autostart do not address this problem, as they presume that Open Shell is already installed.

As long as we're updating the installer, let's think about releasing the MSI directly rather than packing it inside an EXE where the unpacking process could be hijacked by malware watching for such unpacking  procedures.

So +1 for an MSI installer that allows single-user deployment.

**Comment by ge0rdi** [comment]
> Many of us are working on corporate machines where "Install for All User" privileges are disabled. At present, these users cannot install Open Shell for themselves at all.

That's unfortunate, but `Open-Shell` currently needs to do system-wide changes during installation.

For example it needs to register `StartMenuHelper` DLL so that `Open-Shell` Start Menu is running immediately when user logs on.
Without this helper menu starts in cca 10-20 seconds after user logon (when standard `Run` key is processed).

Some parts of `Open-Shell` use HKLM key.
Current installer script also doesn't count with such case.

All these things should be addressed if we want to provide per-user installation.

If you (or anyone) are interested to work on this I can provide some limited guidance (I'm not very familiar with MSI though).

> As long as we're updating the installer, let's think about releasing the MSI directly rather than packing it inside an EXE

Installer does some checks for minimum OS version and it also automatically installs proper version (32/64bit).
Though I agree there should be no problem with providing MSIs too.

**Comment by danielmreck** [comment]
Thank you for identifying the pertinent issues for single-user installs, @ge0rdi. (Awesome handle, BTW.)

> Without this helper menu starts in cca 10-20 seconds after user logon
For someone who simply does not have a choice to do a system-wide install, this 10-20 seconds of inconvenience may be a tolerable compromise vs. not using Open-Shell at all. How many regular users are rebooting so frequently that this would be a problem? Once Open-Shell has started, they are good to go until they shutdown, which may be weeks later.  Perhaps a short-term option could be to offer a single-user install and make clear this caveat.

> I agree there should be no problem with providing MSIs too.
It seems there is no reason not to release the MSIs.  Power users generally know whether they need the 64- or 32-bit version, and it is rather pointless to download the half of the EXE that is assured to be useless on any particular system.

**Comment by ibuprophen1** [comment]
I was looking into a few various items @ge0rdi and found some residual information within the System and Registry when going from the Classic Shell to Open Shell. 

What was odd was the residual information varied depending upon "Upgrading" the Classic Shell to Open Shell Software vs just Uninstalling the Classic Shell and then Installing the Open Shell Software. 

The potential "residual" information (when Classic Shell is/was installed) all depends upon BOTH the "length of time" AND "how much was done" (regarding customizations and such). 

This can/may/might cause some issues that varies from none/unknown (meaning that the individual may not even know/notice it) to issues that end up, in the end, being pointed to the Open Shell Software. 

I hope that I had explained the above information okay... LOL! 

I've been touching base with this and, while trying various scenarios, to see if there's some type of pattern. 

If I can figure this out, maybe I can come up with something in the realm of an additional step to take (to ultimately remove the remaining residual information) after an individual removes/uninstalls the Classic Shell Software. 

... Something along those lines... :-P 



~Ibuprophen

**Comment by ylluminate** [comment]
Any news on this situation? This is an extremely important issue for various users as have been noted. Some such users are students with laptops where online schools require you to operate as a limited user. Having this available would be a big help. 

Out of a sampling of about 2,000 users over the last 2 years, 90% of them prefer Open Shell (Classic Shell) over the new Windows 10 Start Menu. This is especially true for "power users." 

This single-user / limited user functionality, even having to wait 10-20 seconds for initialization at system startup would be very helpful to have for so many people.

**Comment by rkagerer** [comment]
This would still be a very welcome feature.

For anyone else who wants to make Open Shell run for certain users, but not everyone by default, check out ge0rdi's comment here: https://github.com/Open-Shell/Open-Shell-Menu/issues/161#issuecomment-500226960

**Comment by zhmlcg** [comment]
It is end of 2021 now and a few days to 2022. Has the feature of choosing between "Install for All Users" and "Install only for me" implemented? Do we still have to use ge0rdi's trick for this purpose today?
